tmt/utils/__init__.py

```diff
@@ -6347,3 +6348,30 @@ def is_url(url: str) -> bool:
     """ Check if the given string is a valid URL """
     parsed = urllib.parse.urlparse(url)
     return bool(parsed.scheme and parsed.netloc)
+
+
+@contextlib.contextmanager
+def suppress_warning(category: Optional[type[Warning]]) -> Iterator[None]:
```

Comment:
I think we need to handle synchronization, I'm not 100 % sure but I suspect this filtering will not be thread-safe. 
```python
_suppress_warning_lock = threading.RLock()

def suppress_warning(...):
    if category is not None:
        with _suppress_warning_lock:
            try:
                warnings.simplefilter('ignore', category)
                yield
            finally:
                # Reset the warning to its default for a specific category
                warnings.simplefilter('default', category)
    else:
        yield
```

The reentrant lock should allow called code to disable its own warning if necessary, as long as it runs in the same thread.

Reply:
After reading the [docs](https://docs.python.org/3.11/library/warnings.html), I think you are right. It seems that the functions in the `warnings` module are not thread-safe, and we should handle this.

Modified in: https://github.com/teemtee/tmt/pull/3356/commits/029ca37f6e579891fdf6d61b33f3a95afe5ba37f